File: leetcode/easy/Intersection_of_two_arrays2.js
var intersect = function(nums1, nums2) {
    counter = {};
    result = [];
    nums1.forEach((num1) => {
        if (num1 in counter) {
            counter[num1]++;
        } else {
            counter[num1] = 1;
        }
    })
    nums2.forEach((num2) => {
        if (num2 in counter && counter[num2]) {
            result.push(num2);
            counter[num2]--;
        }
    })
    return result;
};

File: leetcode/easy/Intersection_of_two_arrays2.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        n1_counter = collections.Counter(nums1)
        result = []
        for n2 in nums2:
            if n2 in n1_counter and n1_counter[n2]:
                result.append(n2)
                n1_counter[n2] -= 1
        return result
